### PR TITLE
feat: defi opportunity click events

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

- Add mix panel tracking events for clicking on opportunities
- Add helper function to give us composite symbol
- Added the two tokens for dev and prod

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

## Risk

small risk

### Operations

No testing required

### CODE REVIEW

File: src/lib/mixpanel/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { fromAssetId } from '@shapeshiftoss/caip'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { selectAssetById } from 'state/slices/selectors'
+import { store } from 'state/store'
+
+export const GetCompositeAssetSymbol = (assetId?: AssetId) => {
```
Comment (gomesalexandre) [inline src/lib/mixpanel/helpers.ts:7]:
```suggestion
export const getCompositeAssetSymbol = (assetId?: AssetId) => {
```

Comment (gomesalexandre) [inline src/lib/mixpanel/helpers.ts:7]: We should never be calling this with an `undefined` `AssetId` ?

File: src/pages/Defi/components/FeaturedCards/FeaturedCard.tsx [hunk 2]
```diff
@@ -45,6 +49,13 @@ export const FeaturedCard: React.FC<StakingEarnOpportunityType> = ({
   const handleClick = useCallback(() => {
     const { assetNamespace, assetReference } = fromAssetId(assetId)
 
+    mixpanel?.track(MixPanelEvents.ClickOpportunity, {
+      provider,
+      type,
+      assets: underlyingAssetIds.map(assetId => GetCompositeAssetSymbol(assetId)),
```
Comment (gomesalexandre) [inline src/pages/Defi/components/FeaturedCards/FeaturedCard.tsx:55]: Ditto shorthand if you feel like it

File: src/components/EarnDashboard/components/StakingPositions/LpPositions.tsx [hunk 3]
```diff
@@ -66,6 +70,15 @@ export const LpPositions: React.FC<ProviderPositionProps> = ({ ids, assetId }) =
         return
       }
 
+      mixpanel?.track(MixPanelEvents.ClickOpportunity, {
+        provider,
+        type,
+        assets: opportunity.original.underlyingAssetIds.map(assetId =>
+          GetCompositeAssetSymbol(assetId),
+        ),
```
Comment (gomesalexandre) [inline src/components/EarnDashboard/components/StakingPositions/LpPositions.tsx:78]: If you feel like using the shorthand
```suggestion
        assets: opportunity.original.underlyingAssetIds.map(GetCompositeAssetSymbolassetId)
```

**Review by gomesalexandre — COMMENTED** [review]
Comments - other than that looks good to me

File: .env.app [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # feature flags
 REACT_APP_FEATURE_PENDO=true
 
+# mixpanel
+REACT_APP_FEATURE_MIXPANEL=9d304465fc72224aead9e027e7c24356
```
Comment (gomesalexandre) [inline .env.app:5]: Add API_KEY suffix terminology for consistency?

**Review by gomesalexandre — APPROVED** [review]
Conceptually LGTM, added a terminology improvement WRT the env var

File: .env.app (hunk 4 above)
Comment (reallybeard) [reply .env.app:5]: That’s actually totally wrong. The key is token I’ll update.

File: src/lib/mixpanel/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { fromAssetId } from '@shapeshiftoss/caip'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { selectAssetById } from 'state/slices/selectors'
+import { store } from 'state/store'
+
+export const getCompositeAssetSymbol = (assetId: AssetId) => {
+  const asset = selectAssetById(store.getState(), assetId ?? '')
+  const { chainId } = fromAssetId(assetId ?? '')
```
Comment (gomesalexandre) [inline src/lib/mixpanel/helpers.ts:9]:
```suggestion
  const { chainId } = fromAssetId(assetId)
```

**Review by gomesalexandre — APPROVED** [review]

**Pull request merged**